gate-deck-plugins/src/main/kotlin/com/netflix/spinnaker/gate/plugins/DeckPluginCache.kt

```diff
@@ -70,16 +62,15 @@ class DeckPluginCache(
       updateManager.refresh()
 
       val newCache = updateManager.plugins
-        .map {
-          val plugin = DeckPluginVersion(it.id, selectPluginVersion(it))
-          val path = getOrDownload(plugin.id, plugin.version)
-          PluginCacheEntry(plugin, path)
+        .mapNotNull { pluginInfo ->
+          updateManager.getLastPluginRelease(pluginInfo.id, DECK_REQUIREMENT)?.let { lastRelease ->
+            val plugin = DeckPluginVersion(pluginInfo.id, lastRelease.version)
+            val path = getOrDownload(plugin.id, plugin.version)
+            PluginCacheEntry(plugin, path)
+          }
         }
-        .filter {
-          // TODO(rz): Once bundles support manifest artifacts, we can just inspect those
-          //  for a deck plugin rather than downloading everything and seeing if a
-          //  index.js file exists.
-          it.path.resolve("index.js").toFile().exists()
+        .ifEmpty {
+          throw PluginsNotFoundException()
```

Comment:
Changed this to only download plugins that require deck, rather than downloading everything and checking on the existence of an `index.js` file.